code/modules/client/client_colour.dm

```diff
@@ -1,35 +1,21 @@
-#define PRIORITY_ABSOLUTE 1
-#define PRIORITY_HIGH 10
-#define PRIORITY_NORMAL 100
-#define PRIORITY_LOW 1000
+#define CLIENT_COLOR_VALUE_INDEX 1
+#define CLIENT_COLOR_PRIORITY_INDEX 2
 
-/**
- * Client Colour Priority System By RemieRichards (then refactored by another contributor)
- * A System that gives finer control over which client.colour value to display on screen
- * so that the "highest priority" one is always displayed as opposed to the default of
- * "whichever was set last is displayed".
- *
- * Refactored to allow multiple overlapping client colours
- * (e.g. wearing blue glasses under a yellow visor, even though the result is a little unsatured.)
- * As well as some support for animated colour transitions.
- *
- * Define subtypes of this datum
- */
 /datum/client_colour
-	///The color we want to give to the client. This has to be either a hexadecimal color or a color matrix.
-	var/colour
-	///The mob that owns this client_colour.
+	/// Color given to the client, can be a hex color, color matrix or a filter
+	var/color
+	/// The mob that owns this client_colour
 	var/mob/owner
-	/**
-	  * We prioritize colours with higher priority (lower numbers), so they don't get overriden by less important ones:
-	  * eg: "Bloody screen" > "goggles colour" as the former is much more important
-	  */
-	var/priority = PRIORITY_NORMAL
-	///Will this client_colour prevent ones of lower priority from being applied?
+	/// Priority of this color, higher values are rendered above lower ones
+	var/priority = CLIENT_COLOR_FILTER_PRIORITY
+	/// Will this client_colour prevent ones of lower priority from being applied?
 	var/override = FALSE
-	///IF non-zero, 'animate_client_colour(fade_in)' will be called instead of 'update_client_colour' when added.
+	/// If set to TRUE, all colors below and above this one will be rendered in separate filters
+	/// If color is a filter, forced to TRUE
+	var/split_filters = FALSE
+	/// If non-zero, 'animate_client_colour(fade_in)' will be called instead of 'update_client_colour' when added.
 	var/fade_in = 0
-	///Same as above, but on removal.
+	/// Same as above, but on removal.
```

Comment:
feels saner for auto-doc stuff than "refer to this *other* thing"
```suggestion
	/// If non-zero, 'animate_client_colour(fade_out)' will be called instead of 'update_client_colour' when removed.
```
suggestion to just replace it with an updated version of the comment it's referring to